File: 백준 2/BJ_1072/BJ_1072.py
import sys
input = sys.stdin.readline

x, y = map(int, input().split())

# 부동소수점 오류 생각해서 int() 쓰지 말고 // 사용. *100 순서도 중요
z = (y * 100) // x
temp = z

# 승률이 99 이상이면 바로 -1을 출력 아... 99... 멍청한 나
if z < 99 :
    left = y
    right = 1000000000*2
    mid = 0
    while left + 1 < right:
        mid = (left + right) // 2
        # 승률 재계산 총 이긴 횟수/총 게임 횟수 * 100
        temp = (mid * 100) // (x + mid - y)
        
        if temp == z:
            left = mid
        else: 
            right = mid
            
    print(right - y)
else:
    print(-1)

File: 백준/~2024/Binary_search/BJ_1072/BJ_1072.cpp
#include <bits/stdc++.h>
using namespace std;
#define X_BORDER 1000000000

int main(int argc, char const *argv[])
{
    ios::sync_with_stdio(0); cin.tie(NULL); cout.tie(NULL);

    // 입력받기
    unsigned long long x; cin >> x;
    unsigned long long y; cin >> y;
    unsigned long long z = (unsigned long long) (y*100 / x);

    // <알고리즘 ~ 정답출력>
    if(z >= 99)
        cout << -1 << "\n";
    else
    {
        unsigned long long left = y;
        unsigned long long right = X_BORDER*2;

        while(left + 1 < right)
        {
            unsigned long long mid = (unsigned long long) (left + right) / 2;
            unsigned long long temp = (unsigned long long) (mid*100 / (x + mid - y));
            if(temp == z)
                left = mid;
            else
                right = mid;
        }
        cout << right - y << "\n";
    }
    
    return 0;
}